scripts/release/asset_publish.py

```diff
@@ -221,34 +240,59 @@ def assemble_command(
     if debug_mode:
         cmd.append("--debug")
     print(cmd)
-
     return cmd
 
 
-def run_command(
-    cmd, failure_list: List, debug_mode: bool = None
+def publish_asset(
+    asset,
+    registry_name,
+    resource_group,
+    workspace_name,
+    version,
+    failure_list,
+    debug_mode: bool = None
 ):
-    """Run the az cli command for pushing the model to registry."""
+    """Publish asset to registry."""
+    cmd = asset_publish_command(
+        asset.type.value, str(asset.spec_with_path),
+        registry_name, version, resource_group, workspace_name, debug_mode
+    )
+
+    # Run command
+    result = run_command(cmd)
     if debug_mode:
         # Capture and redact output
-        results = run(
-            cmd,
-            stdout=PIPE,
-            stderr=STDOUT,
-            encoding=sys.stdout.encoding,
-            errors="ignore",
-        )
-        redacted_output = re.sub(r"Bearer.*", "", results.stdout)
-        logger.print(redacted_output)
-    else:
-        results = run(cmd)
+        redacted_output = re.sub(r"Bearer.*", "", result.stdout)
+        redacted_err = re.sub(r"Bearer.*", "", result.stderr)
+        print(f"stdout: {redacted_output}")
+        print(f"stderr: {redacted_err}")
```

Comment:
A little more readable:
```suggestion
        if redacted_output:
            print(f"STDOUT:\n{redacted_output}")
        if redacted_err:
            print(f"STDERR:\n{redacted_err}")
```